src/fsharp/LexFilter.fs

```diff
@@ -677,105 +704,119 @@ type LexFilterImpl (lightStatus: LightSyntaxStatus, compilingFsLib, lexer, lexbu
     // Undentation rules
     //--------------------------------------------------------------------------
 
+    let (|RelaxWhitespace2|_|) x = if lexbuf.SupportsFeature LanguageFeature.RelaxWhitespace2 then Some x else None
     let pushCtxt tokenTup (newCtxt: Context) =
         let rec undentationLimit strict stack = 
             match newCtxt, stack with 
             | _, [] -> PositionWithColumn(newCtxt.StartPos, -1) 
 
             // ignore Vanilla because a SeqBlock is always coming 
             | _, (CtxtVanilla _ :: rest) -> undentationLimit strict rest
-
-            | _, (CtxtSeqBlock _ :: rest) when not strict -> undentationLimit strict rest
-            | _, (CtxtParen _ :: rest) when not strict -> undentationLimit strict rest
+            | (_, (CtxtSeqBlock _ :: rest) | _, (CtxtParen _ :: rest)) when not strict
```

Comment:
Please remove any changes that are just cleanup.  To assess this we need a PR with an absoltuely minimal diff